### ISSUE TITLE
Map constantly re renders

### ISSUE THREAD

**Issue by KristenKatona** [venits/react-native-map-clustering#143]
I plugged in your wonderful `MapView` in place of one where I was using `MapView` from `react-native-maps` and it stopped using my current location on initial load. We were able to fix that by updating the `onUserLocationChange` callback and everything seemed fine on the iOS simulator and the Android emulator. BUT on a physical Android phone, the map continually re-renders and the `onUserLocationChange` constantly fires off with slightly different coordinates each time. Any idea what might be causing it to re-render continually and shift the coordinates every so slightly?

Here's the `props` that we pass into the `MapView`:
```
children: (50) [{…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}]
clusterColor: "#022b6e"
clusterTextColor: "#FCBA00"
initialRegion: {
    latitude: 39.6515089
    latitudeDelta: 0.5
    longitude: -79.9480251
    longitudeDelta: 0.5
}
key: "{"latitude":39.6515089,"longitude":-79.9480251,"latitudeDelta":0.5,"longitudeDelta":0.5}"
minZoomLevel: 7
onRegionChangeComplete: ƒ (region)
onUserLocationChange: ƒ ()
provider: "google"
radius: 100
ref: {current: null}
showsMyLocationButton: true
showsUserLocation: true
style: {flex: 1, minHeight: 472.71428571428567}
```

Not sure if this is related so I'll mention it just in case - I see a warning in the console as well that comes from `ClusteredMapView`: 
```
ExceptionsManager.js:82 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
    in ClusteredMapView
    in ClusteredMapView (at Map/index.js:41)
    in Map
    in Map (at UserMap/index.js:41)
```

**Comment by Baterka** [comment]
I think I am also experiencing this. I have about 2.5k markers and map gets slower and slower until crashes on memory.

**Comment by ruslanfraps** [comment]
> I think I am also experiencing this. I have about 2.5k markers and map gets slower and slower until crashes on memory.

try tracksViewChanges={false} for your markers

**Comment by nashashibi** [comment]
problem is tracksViewChanges is not being explicitly passed down to the <Marker /> component internally in the library.

You need to do yourself for now. You can either fork it or just edit the package in node_modules directly (I did the latter). 

Just go to ClusteredMapView.js and ClusteredMarker.js and find the <Marker> component and add tracksViewChanges={false}

**Comment by rezasazesh** [comment]
@nashashibi Thank you! This solves the issue! Would you be so kind to submit a PR?

**Comment by nashashibi** [comment]
@rezasazesh I was planning to do this soon but apparently there's an open PR for this fix already.

**Comment by rezasazesh** [comment]
@nashashibi I saw it, its amazing!!

**Comment by venits** [comment]
Fixed in module version `3.2.0`

Thank you for your contribution ❤️

**Comment by JorchCortez** [comment]
> problem is tracksViewChanges is not being explicitly passed down to the `<Marker />` component internally in the library.
> 
> You need to do yourself for now. You can either fork it or just edit the package in node_modules directly (I did the latter).
> 
> Just go to **ClusteredMapView.js** and **ClusteredMarker.js** and find the `<Marker>` component and add `tracksViewChanges={false}`

OMG, 4 years after this actually saved my butt today!
